_posts/2018-3-11-interactive-examples.md

```diff
@@ -0,0 +1,100 @@
+
+Over the last year and a bit, the [MDN Web Docs] team has been designing, building, and implementing interactive examples for our reference pages. The motivation for this was the idea that MDN should do more to help "action-oriented" users: people who like to learn by seeing and playing around with example code, rather than by reading about it.
+
+We've just finished adding interactive examples for the JavaScript and CSS reference pages. This post looks back at the project to see how we got here and what we learned on the way.
+
+## First prototypes
+
+The project began as a suggestion by [Florian Scholz], at an MDN meetup in December 2016, that we should add simple examples to the tops of MDN reference pages. Initially we didn't plan to make them interactive: we thought that they would just be static code samples. It was I think [Julien Gattelier] and [Sebastian Zartner] who first suggested that they could be more interesting, with an editor that provided autocomplete suggestions. So I incorporated the wonderful [Code Mirror], and we built some prototypes to test.
+
+In the early prototypes the JS editor looked something like this:
+
+![](../images/interactive-examples/js-editor-first-user-test.png)
+
+The CSS editor looked something like this:
+
+![](../images/interactive-examples/css-editor-first-user-test.png)
+
+## First round of user testing
+
+In March 2017 [Kadir Topal] and I attended the first round of user testing, which was run by [Mark Hurst]. We learned a great deal about user testing, about our prototypes, and about what users wanted to see. We learned that users usually wanted to see examples and appreciated them being quick to find. Users liked interactive examples, too. But *noone* understood how to work our autocomplete, so reluctantly we let it go.
+
+Especially for CSS, though, we still wanted a way to show users the different kinds of syntax that an item can accept. For the CSS pages, we already had a PRE block in the pages listing syntax options, like this:
+
+    transform: matrix(1.0, 2.0, 3.0, 4.0, 5.0, 6.0);
+    transform: translate(12px, 50%);
+    transform: translateX(2em);
+    transform: translateY(3in);
+    transform: scale(2, 0.5);
+    transform: scaleX(2);
+    transform: scaleY(0.5);
+    transform: rotate(0.5turn);
+    transform: skew(30deg, 20deg);
+
+One user interaction we saw, that we really liked, was when users would copy lines from this PRE block into the editor, to see the effect. So we thought of combining this PRE block and the editor:
+
+![](../images/interactive-examples/css-editor-first-of-final.png)
+
+In this version, you can select a line from the block underneath, and the style is applied to the element above.
+
+It looks terrible now, but it's interesting that the basic interaction model that we would eventually use for both JavaScript and CSS editors was already in place.
```

Comment:
I think this is an interesting point in an interesting section that you could make a tad more explicit - the fact that you weren't really sure what sort of interactivity would be engaging to a site visitor, and you used the user testing sessions to inform the design process....

Reply:
I've had a go at expanding this and also used GIFs to try to show the interaction better. If you think this is a good idea I'll try to make some better-looking GIFs for the actual post.